airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py

```diff
@@ -251,6 +251,7 @@ def should_retry(self, response: requests.Response) -> bool:
             self.logger.warning(f"Caught 40002: {data}, {response.url}, {self.retried_40002_counter}")
             self.retried_40002_counter += 1
             return True
+        self.retried_40002_counter = 0
```

Comment:
Good catch! So that's why it wasn't logging properly and failing on first try. Just updated it. I'd rather keep both modifications, because it does not make sense to me to have a "global counter" on this. Especially since our sync is quite long (hours).

Reply:
Right, but then we could just remove the counter altogether since we're resetting it every time it called.

On the other hand, googling 40002 I have a suspicion this is a generic validation/access error so I wouldn't retry on it universally.

Reply:
I just pushed [this ](https://github.com/VoodooTeam/airbyte/pull/141/commits/8f210fee0d382040735f05c4e61903dde921f2a2)

Reply:
👍